FLIR/conservator/managers/media.py

```diff
@@ -1,3 +1,257 @@
+import functools
+import logging
+import multiprocessing
+import os
+import time
+
+from FLIR.conservator.fields_request import FieldsRequest
+from FLIR.conservator.util import upload_file
+from FLIR.conservator.managers.searchable import AmbiguousIdentifierException
+from FLIR.conservator.wrappers.media import MediaType, MediaUploadRequest, MediaCompare
+from FLIR.conservator.wrappers.queryable import InvalidIdException
+
+logger = logging.getLogger(__name__)
+
+
+class ProcessingTimeoutError(TimeoutError):
+    """
+    Raised when the amount of time spent waiting for media to process exceeds
+    the requested timeout.
+    """
+
+    pass
+
+
 class MediaTypeManager:
     def __init__(self, conservator):
         self._conservator = conservator
+
+    def from_path(self, string, fields="id"):
+        if not "/" in string:
+            return None
+
+        # start by path lookup
+        parent_path = "/".join(string.split("/")[:-1])
+        name = string.split("/")[-1]
+
+        parent = self._conservator.collections.from_remote_path(
+            path=parent_path, make_if_no_exist=False, fields="id"
+        )
+
+        # look inside parent for media with exact name match
+        fields = FieldsRequest.create(fields)
+        fields.include_field("name")
+        media = list(parent.get_media(fields=fields, search_text="name:" + name))
```

Comment:
PaginatedQuery can filter for exact name match

```suggestion
        media = list(parent.get_media(fields=fields, search_text="name:" + name).filtered_by(name=name))
```

Reply:
This is an example of media not being so parallel to the other classes. The return of get_media() isn't actually a `PaginatedQuery` but rather a `generator` that will first iterate through results of the video query and then through results of the image query, so it's too late to add a filtered_by() at that point.

Reply:
Ah yes you're right. Hopefully we can come up with a more elegant solution to the Media queries